File: minify.js
const fs = require('fs');
const path = require('path');

const filePath = './stylique-virtual-tryon/shopify/Shopify_new_tryon_upload_first.liquid';

let content = fs.readFileSync(filePath, 'utf-8');
const originalSize = content.length;

console.log(`Original file size: ${originalSize} bytes (${(originalSize/1024).toFixed(2)} KB)`);

// Step 1: Remove Liquid comments
content = content.replace(/{%\s*comment\s*%}[\s\S]*?{%\s*endcomment\s*%}/gi, '');
console.log(`After removing Liquid comments: ${content.length} bytes`);

// Step 2: Remove HTML comments
content = content.replace(/<!--[\s\S]*?-->/g, '');
console.log(`After removing HTML comments: ${content.length} bytes`);

// Step 3: Remove CSS comments
content = content.replace(/\/\*[\s\S]*?\*\//g, '');
console.log(`After removing CSS comments: ${content.length} bytes`);

// Step 4: Minify CSS blocks
content = content.replace(/<style[^>]*>([\s\S]*?)<\/style>/gi, (match, css) => {
  css = css.replace(/\n/g, ' ');  // Remove newlines
  css = css.replace(/\s*([{}:;,>+~])\s*/g, '$1');  // Remove spaces around operators
  css = css.replace(/\s+/g, ' ');  // Multiple spaces to single
  return `<style>${css.trim()}</style>`;
});
console.log(`After minifying CSS blocks: ${content.length} bytes`);

// Step 5: Minify JS blocks
content = content.replace(/<script[^>]*>([\s\S]*?)<\/script>/gi, (match, js) => {
  js = js.replace(/\n/g, ' ');  // Remove newlines
  js = js.replace(/\s*([=+\-*/%<>!&|^?:;,()[\]{}])\s*/g, '$1');  // Remove spaces around operators
  js = js.replace(/\s+/g, ' ');  // Multiple spaces to single
  js = js.replace(/\s*\(/g, '(');  // Remove space before (
  return `<script>${js.trim()}</script>`;
});
console.log(`After minifying JS blocks: ${content.length} bytes`);

// Step 6: Remove multiple spaces
content = content.replace(/  +/g, ' ');
console.log(`After removing multiple spaces: ${content.length} bytes`);

// Step 7: Remove blank lines
content = content.replace(/\n\s*\n+/g, '\n');
console.log(`After removing blank lines: ${content.length} bytes`);

// Step 8: Remove trailing whitespace on lines
content = content.split('\n').map(line => line.trimRight()).join('\n');
console.log(`After removing trailing spaces: ${content.length} bytes`);

// Step 9: Remove space before >
content = content.replace(/\s+>/g, '>');
console.log(`After removing space before >: ${content.length} bytes`);

// Step 10: Minify inline styles
content = content.replace(/style="([^"]*)"/g, (match, style) => {
  style = style.replace(/\s*([{}:;,])\s*/g, '$1');
  return `style="${style}"`;
});
console.log(`After minifying inline styles: ${content.length} bytes`);

// Write back
fs.writeFileSync(filePath, content, 'utf-8');

const finalSize = content.length;
const reduction = originalSize - finalSize;
const reductionPct = ((reduction / originalSize) * 100).toFixed(1);

console.log(`\n=== MINIFICATION COMPLETE ===`);
console.log(`Final file size: ${finalSize} bytes (${(finalSize/1024).toFixed(2)} KB)`);
console.log(`Reduction: ${reduction} bytes (${reductionPct}%)`);
console.log(`File successfully minified and written back!`);

File: minify.py
import re
import os

file_path = r"stylique-virtual-tryon/shopify/Shopify_new_tryon_upload_first.liquid"

# Read the file
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

original_size = len(content.encode('utf-8'))
print(f"Original size: {original_size / 1024:.1f} KB")

# Step 1: Remove CSS comments /* ... */
content = re.sub(r'/\*[\s\S]*?\*/', '', content)
print("Step 1: CSS comments removed")

# Step 2: Remove JS comments // ... (but be careful with URLs)
# Only remove // comments that are on their own or at the end of a line
content = re.sub(r'//.*?(?=\n)', '', content)
print("Step 2: JS comments removed")

# Step 3: Condense multiple spaces to single space
content = re.sub(r'[ \t]+', ' ', content)
print("Step 3: Multiple spaces condensed")

# Step 4: Remove newlines except where needed
# Replace multiple newlines with single newline first
content = re.sub(r'\n\s*\n+', '\n', content)
content = re.sub(r'\n[ \t]*([{}:;,])', r'\1', content)
content = re.sub(r'([{}:;,])\n[ \t]*', r'\1', content)
print("Step 4: Newlines condensed")

# Step 5: Remove blank lines
content = re.sub(r'^\s*\n', '', content, flags=re.MULTILINE)
print("Step 5: Blank lines removed")

# Step 6: Remove spaces around CSS braces and semicolons
content = re.sub(r'\s*([{}:;])\s*', r'\1', content)
print("Step 6: CSS punctuation spacing removed")

# Step 7: Additional aggressive minification - remove spaces around = and , in specific contexts
content = re.sub(r'\s*([\(\)\[\]=,])\s*', r'\1', content)
print("Step 7: Additional punctuation spacing removed")

new_size = len(content.encode('utf-8'))
reduction = original_size - new_size
reduction_percent = (reduction / original_size) * 100

print(f"\nNew size: {new_size / 1024:.1f} KB")
print(f"Reduction: {reduction / 1024:.1f} KB ({reduction_percent:.1f}%)")
print(f"Target achieved: {new_size < 220 * 1024}")

# Write back the minified content
with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print("\nFile has been minified and saved!")
